lib/compress/huf_compress.c

```diff
@@ -357,16 +356,17 @@ size_t HUF_buildCTable_wksp (HUF_CElt* tree, const unsigned* count, U32 maxSymbo
     int lowS, lowN;
     U16 nodeNb = STARTNODE;
     U32 nodeRoot;
+    rankPos* rank = (rankPos *)(huffNode0 + HUF_CTABLE_WORKSPACE_SIZE / sizeof(nodeElt*));
```

Comment:
Looks like our contbuild is failing on this line, due to potential aliasing issues. You'll need to experiment to confirm, but I imagine you can fix this by casting through `char*` or `void*`.

Maybe:

```
rankPos* rank = (rankPos*)((BYTE*)workspace + HUF_CTABLE_WORKSPACE_SIZE);
```

You should be able to check with `make clean zstd MOREFLAGS="-Wstrict-aliasing -Werror"`.

Reply:
It might also make more sense to give this variable a slightly more descriptive name now that its been hoisted out of `HUF_sort`.

Reply:
> Looks like our contbuild is failing on this line, due to potential aliasing issues

Indeed, strict typing will not allow a `rankPos*` to be derived from a `nodeElt*`.